File: php/app/leetcode/s0003_longest_substring_without_repeating_characters/SolutionTest.php
<?php

use App\LeetCode\s0003_longest_substring_without_repeating_characters\Solution;
use Yosymfony\Toml\Toml;

$solution = new Solution();

function readTestFile(): array
{
    $cases = [];
    $test = Toml::parseFile(__DIR__ . "/../../../../test.toml");

    for ($i = 0; $i < count($test["table3"]["expected"]); $i++) {
        $cases["示例" . ($i + 1)] = [
            "expected" => $test["table3"]["expected"][$i],
            "s" => $test["table3"]["s"][$i],
        ];
    }

    return $cases;
}

test("3.无重复字符的最长子串", function (int $expected, string $s) use ($solution): void {
    expect($solution->lengthOfLongestSubstring($s))->toEqual($expected);
})->with(readTestFile());

File: golang/src/leetcode/s0003_longest_substring_without_repeating_characters/solution_test.go
package s0003_longest_substring_without_repeating_characters

import (
	"github.com/BurntSushi/toml"
	"github.com/stretchr/testify/assert"
	"log"
	"testing"
)

type Test struct {
	Table3 Owner `toml:"table3"`
}

type Owner struct {
	Expected []int    `toml:"expected"`
	S        []string `toml:"s"`
}

type Case struct {
	expected int
	s        string
}

var cases []Case

func readTestFile() {
	var test Test

	if _, err := toml.DecodeFile("../../../../test.toml", &test); err != nil {
		log.Fatal(err)
	}

	for i := 0; i < len(test.Table3.Expected); i++ {
		cases = append(cases, Case{
			expected: test.Table3.Expected[i],
			s:        test.Table3.S[i]},
		)
	}
}

func TestLengthOfLongestSubstring(t *testing.T) {
	readTestFile()

	for _, c := range cases {
		t.Run("3.无重复字符的最长最长", func(t *testing.T) {
			assert.Equal(t, c.expected, lengthOfLongestSubstring(c.s))
		})
	}
}
